lib/topics_filter.rb

```diff
@@ -77,6 +81,14 @@ def filter_status(status:, category_id: nil)
 
   private
 
+  def filter_by_number_of_posts(min: nil, max: nil)
+    value = min || max
+    return if value !~ /^\d+$/
+
+    @scope = @scope.where("topics.posts_count >= ?", value) if min
+    @scope = @scope.where("topics.posts_count <= ?", value) if max
+  end
```

Comment:
It's possible to call it with an invalid max value:

```ruby
def filter_by_number_of_posts(min: nil, max: nil)
  value = min || max
  return if value !~ /^\d+$/

  p "MIN CALLED: #{min}" if min
  p "MAX CALLED: #{max}" if max
end

filter_by_number_of_posts(min: "1", max: "foo")
# "MIN CALLED: 1"
# "MAX CALLED: foo"
```

It's also a little bit unexpected that min/max need to be strings as min/max are generally integers, although, not critical.

Reply:
> It's possible to call it with an invalid max value which in our case would make the query wrong:

Good catch, I've fixed it.